modules/home/products/SearchAndFilterProducts.tsx

```diff
@@ -32,55 +43,40 @@ const SearchAndFilterProducts = (prop: {
   const [activeSection, setActiveSection] = useState(0);
   const [selectedOption, setSelectedOption] = useState<string>('');
   const [selectedOption2, setSelectedOption2] = useState<string>('');
-  const [showLeftButton, setShowLeftButton] = useState<boolean>(true);
-  const [showRightButton, setShowRightButton] = useState<boolean>(true);
+  const [showLeftButton, setShowLeftButton] = useState<boolean>(false);
   const [showFilterComponent, setShowFilterComponent] = useState<boolean>(false);
 
   const { filters, handleFilters } = prop;
   const sliderRef = useRef<HTMLDivElement>(null);
 
-  const handleScroll = () => {
-    const slider = sliderRef.current!; // Non-null assertion
-
-    // setShowLeftButton(!isStart);
-    // setShowRightButton(!isEnd);
-  };
-
-  const closeFilterComponent = (option?: 'close' | 'clear') => {
-    if (option === 'clear') {
-      prop.setFilter({});
-    } else {
-      setShowFilterComponent(false);
-    }
-  };
   const slideLeft = () => {
-    const slider = sliderRef.current!; // Non-null assertion
-    slider.scrollLeft -= 150; // Adjust the scroll distance as needed
+    const slider = sliderRef.current!;
+    slider.scrollLeft -= 150;
   };
 
   const slideRight = () => {
-    const slider = sliderRef.current!; // Non-null assertion
-    slider.scrollLeft += 150; // Adjust the scroll distance as needed
+    setShowLeftButton(true);
+    const slider = sliderRef.current!;
```

Comment:
confirm that slider ref is not null instead

Reply:
fixed